### DISCUSSION TITLE
Read-Only Mode for alarm status

### DISCUSSION THREAD

**Discussion by Tycho-S** [tsightler/ring-mqtt#517] [Feature Requests]
Would it be possible to create a Read-Only mode for the integration? E.g. read the status but not allow arm/disarm at all.

I would really like to have the integration (and thus home assistant) to be able to monitor the status of my alarm, but not the capability to arm/disarm it. Why? Well, it greatly increases the attack surface of my alarm to include the integration, my MQTT server, and home assistant as well. An attacker has many more points to mess with it and possibly generate a disarm event.

It would be great if we could connect to the Ring API but only receive access to read events and not change them. I always arm/disarm the alarm with the panel anyway and I'd like to keep it that way for the above reason.

**Comment by tsightler** [comment]
The code is open, you can do what you like, but I have no plans to implement this.  It might work if you just give it a guest user role in Ring, but I've never tested this and I have no idea if it would be functional or not.

**Comment by Tycho-S** [comment]
Ok yes I can try that thanks! That would be great if it does work as ring would disallow it on their side , making circumvention difficult. 

All I want to do in HA is monitor the alarm status and trigger some actions (eg turn off all the lights when I arm the alarm). And perhaps monitor individual sensors.

I’m surprised there’s not been more people asking for this due to the sensitive nature of the alarm disarming. But I could look at that myself indeed if guest mode doesn’t work. I’m a bit paranoid because I’m in cybersecurity but my building has been burgled 3 times in the last year :)

Thanks in any case for all your work on this app!

**Reply by tsightler** [reply]
Lots of people build homebrew alarms using HA, using tools like Alarmo or others, so, if people are OK with HA being their entire alarm system, I think they are probably OK with disarming the alarm.  Realistically, the entire reason I initially wrote the very first version of ring-mqtt was to automate disarming of the alarm at a given time each morning if it was in armed-home mode, so my wife/daughter wouldn't accidentally set it off when leaving for school.  It was a feature that my old alarm had for more than a decade, but that I still don't think exist in the Ring app.

The problem is anything I would be able to build in ring-mqtt would have to be configurable and thus easily modified by an attacker as well, so it's difficult to see how this would add much in the way of security.

However, you could at least make it a little bit more difficult for an attacker by using MQTT access control to block writes to the command topic for the alarm.  If you use the local MQTT broker via an addon then an attacker that gets access to HA would be able to modify this, however, you can always run an external MQTT broker on another machine, with no other access, and set very granular access permissions on the MQTT topics.  That way, you could block arming/disarming commands at the MQTT broker, and an attacker would have to hack another system to change the permissions.

For even an additional layer, you could run ring-mqtt with its own broker that listened only on 127.0.0.1, and then configure that broker to forward only the events you want to HA.

In other words, you have lots of options to add layers of security that would make such an attack difficult, without any changes to ring-mqtt itself.
